tensorflow_io/cifar/python/ops/cifar_dataset_ops.py

```diff
@@ -0,0 +1,148 @@
+from tensorflow import dtypes
+from tensorflow.compat.v1 import data
+from tensorflow_io import _load_library
+cifar_ops = _load_library('_cifar_ops.so')
+
+class _CIFAR10Dataset(data.Dataset):
+  """A CIFAR File Dataset that reads the cifar file."""
+
+  def __init__(self, filename):
+    """Create a `CIFARDataset`.
+
+    Args:
+      filename: A `tf.string` tensor containing one or more filenames.
+    """
+    self._data_input = cifar_ops.cifar10_input(filename)
+    super(_CIFAR10Dataset, self).__init__()
+
+  def _inputs(self):
+    return []
+
+  def _as_variant_tensor(self):
+    return cifar_ops.cifar10_dataset(
+        self._data_input,
+        output_types=self.output_types,
+        output_shapes=self.output_shapes)
+
+  @property
+  def output_classes(self):
+    return tensorflow.Tensor, tensorflow.Tensor
+
+  @property
+  def output_shapes(self):
+    return (tensorflow.TensorShape([]), tensorflow.TensorShape([3, 32, 32]))
+
+  @property
+  def output_types(self):
+    return dtypes.uint8, dtypes.uint8
+
+class CIFAR10Dataset(_CIFAR10Dataset):
+  """A CIFAR File Dataset that reads the cifar file."""
+
+  def __init__(self, filename, test=False):
+    """Create a `CIFAR10Dataset`.
+
+    Args:
+      filename: A `tf.string` tensor containing one or more filenames.
+    """
+    if test:
+      self._filename = filename + "|test_batch.bin"
+    else:
+      entries = ["|data_batch_" + str(e) + ".bin" for e in range(1, 6)]
+      self._filename = [filename + entry for entry in entries]
+    super(CIFAR10Dataset, self).__init__(self._filename)
+
+  def _as_variant_tensor(self):
+    return _CIFAR10Dataset(self._filename).map( # pylint: disable=protected-access
+        lambda label, image: (tensorflow.transpose(image, [1, 2, 0]), label))._as_variant_tensor()
+
+  @property
+  def output_shapes(self):
+    return (tensorflow.TensorShape([32, 32, 3]), tensorflow.TensorShape([]))
+
+class _CIFAR100Dataset(data.Dataset):
+  """A CIFAR File Dataset that reads the cifar file."""
+
+  def __init__(self, filename):
+    """Create a `CIFARDataset`.
+
+    Args:
+      filename: A `tf.string` tensor containing one or more filenames.
+    """
+    self._data_input = cifar_ops.cifar100_input(filename)
+    super(_CIFAR100Dataset, self).__init__()
+
+  def _inputs(self):
+    return []
+
+  def _as_variant_tensor(self):
+    return cifar_ops.cifar100_dataset(
+        self._data_input,
+        output_types=self.output_types,
+        output_shapes=self.output_shapes)
+
+  @property
+  def output_classes(self):
+    return tensorflow.Tensor, tensorflow.Tensor, tensorflow.Tensor
+
+  @property
+  def output_shapes(self):
+    return (tensorflow.TensorShape([]),
+            tensorflow.TensorShape([]),
+            tensorflow.TensorShape([3, 32, 32]))
+
+  @property
+  def output_types(self):
+    return dtypes.uint8, dtypes.uint8, dtypes.uint8
+
+class CIFAR100Dataset(_CIFAR100Dataset):
+  """A CIFAR File Dataset that reads the cifar file."""
+
```

Comment:
Are we expecting other types than train and test? If so, we can introduce new argument for different types